### PR TITLE
feat(rome_diagnostics): add `with_severity` API

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

This PR adds a new API to `DiagnosticExt` called `with_severity`, which allows changing the severity of a diagnostic in a "native" way. 

This change allows removing an API I had to create in the `AnalyzerDiagnostic`, which is not needed anymore.

## Test Plan

This is a "under the hoods" change, which means that the existing tests should pass.

### CODE REVIEW

File: crates/rome_diagnostics/src/context.rs [hunk 1]
```diff
@@ -42,6 +42,11 @@ pub trait DiagnosticExt: internal::Sealed + Sized {
     fn with_tags(self, tags: DiagnosticTags) -> Error
     where
         Error: From<internal::TagsDiagnostic<Self>>;
+
+    /// Returns a new diagnostic with additional `severity`
+    fn with_severity(self, severity: Severity) -> Error
```
Comment (leops) [inline crates/rome_diagnostics/src/context.rs:47]: For feature parity, can we add this new method to the `Context` trait too ? (It looks like I forgot to add a few methods from `DiagnosticExt` in that trait already already, but in theory they should be kept in sync)

File: crates/rome_diagnostics/src/context.rs [hunk 2]
```diff
@@ -528,7 +543,7 @@ mod internal {
     }
 
     /// Diagnostic type returned by [super::DiagnosticExt::with_tags],
-    /// ùerges `tags` with the tags of its source
+    /// replaces `tags` with the tags of its source
```
Comment (leops) [inline crates/rome_diagnostics/src/context.rs:546]: Nit: While there is a typo here, it is actually merging the two sets of tags and not just replacing them

File: crates/rome_analyze/src/diagnostics.rs [hunk 3]
```diff
@@ -157,13 +157,6 @@ impl AnalyzerDiagnostic {
         }
     }
 
-    /// Sets the severity of the current diagnostic
-    pub fn set_severity(&mut self, new_severity: Severity) {
```
Comment (leops) [inline crates/rome_analyze/src/diagnostics.rs:161]: The `severity` field on `DiagnosticKind::Rule` could be removed if it isn't being set anywhere anymore

File: crates/rome_diagnostics/src/context.rs (hunk 1 above)
Comment (ematipico) [reply crates/rome_diagnostics/src/context.rs:47]: I also added `with_tags`

**Review by leops — APPROVED** [review]

**Pull request merged**